### DISCUSSION TITLE
Import pdf's?

### DISCUSSION THREAD

**Discussion by seanap** [TandoorRecipes/recipes#263] [Q&A] [answered]
Hello, amazing project you have here thank you for sharing.  My wife has a folder on our nas filled with pdf's and cookbooks.  Is there a way to import these pdf's or pages from the epub cookbooks?

**Comment by vabene1111** [comment, marked as answer]
Yes, this is what this application was originally created for (my personal PDF collection).

You will have to put all your recipes into either a Nextcloud or Dropbox. Then add a new storage source and create a sync path. 

Sadly only Dropbox and Nextcloud are supported atm. but others could be added. Feel free to do so yourself or open an issue with the desired integration and maybe i will get around to implementing it.

**Reply by jschwalbe** [reply]
Can you go into more detail on this? My wife also has folders of PDFs and I would love to get them into one structured database. From what I can tell, that is not yet possible, but your answer makes it seem that it is possible. Can you clarify please? Thanks!

**Reply by vabene1111** [reply]
its a bit outdated from a menu strucutre perspective but the principles still hold true https://docs.tandoor.dev/features/external_recipes/

I think also dropbox does not work anymore since they changed their API but nextcloud still works.

**Comment by pippo73** [comment]
Hello, I'm sorry if I resume this old request.
Time is passed since the first request and the new version is been released.
I would like to add a pdf but I cannot find where to add the external storage in the Tandoor 2 version?
In the docs https://docs.tandoor.dev/features/external_recipes/ is indicated that I should find somewhere a way to add a dropbox dir, but I cannot find where is this feature.
Could someone help me pls?

**Reply by vabene1111** [reply]
its under database > Storage

the documentation is sadly a bit outdated and still for tandoor 1 but everything works in tandoor 2 as well

**Reply by pippo73** [reply]
Thank you I will take a look. :-)

Just one quest, in the documentation you talk about a **Username**. Do you mean the email ?

**Reply by vabene1111** [reply]
it depends on the provider, some use usernames some email I think

**Comment by Bootta11** [comment]
Can tandoor extract multiple recipes from pdf cookbook?

**Reply by vabene1111** [reply]
no, currently it only works with one recipe per file

**Comment by carllvin** [comment]
I wrote this tool to help me with this: https://github.com/carllvin/cookbook-importer
